addons/l10n_sa_edi/models/account_edi_format.py

```diff
@@ -443,9 +443,13 @@ def _check_move_configuration(self, invoice):
             )
         if invoice.invoice_date > fields.Date.context_today(self.with_context(tz='Asia/Riyadh')):
             errors.append(_("- Please, make sure the invoice date is set to either the same as or before Today."))
-        if invoice.move_type in ('in_refund', 'out_refund') and not invoice._l10n_sa_check_refund_reason():
+
+        if invoice.l10n_sa_show_reason and not invoice.l10n_sa_reason:
+            errors.append(
+                _("- Please make sure the 'ZATCA Reason' for the issuance of the Credit/Debit Note is specified."))
```

Comment:
Not done ? Either one line or at least put the parenthesis on the next line 
```suggestion
            errors.append(
                _("- Please make sure the 'ZATCA Reason' for the issuance of the Credit/Debit Note is specified.")
            )
```